### PR TITLE
Implement SQL function left

### LINKED ISSUE #7333: Implement support for `left(string, int)`
Implement support for left-trimming string as a SQL function, similarly to what Postgres provides.

### PR DESCRIPTION
**Pull request by antiguru**
Fixes #7333.

Signed-off-by: Moritz Hoffmann <mh@materialize.com>

### CODE REVIEW

File: test/sqllogictest/string.slt [hunk 1]
```diff
@@ -939,6 +939,128 @@ SELECT position(e'\u0903\u0951' IN e'\u0930\u0903\u0951')
...
+----
+रः
+
+# this is exactly the same as above, but using unicode escapes
+query T
+SELECT left(e'\u0930\u0903\u0951', -1)
+----
+रः
+
+# integer edge cases
```
Comment (aljoscha) [inline test/sqllogictest/string.slt:1046]: Did you check whether these match what Postgres would be doing?

File: test/sqllogictest/string.slt [hunk 2]
```diff
@@ -939,6 +939,128 @@ SELECT position(e'\u0903\u0951' IN e'\u0930\u0903\u0951')
...
+SELECT left(vccol, 0) FROM lefttest
+----
+(empty)
+(empty)
+NULL
+(empty)
+(empty)
+(empty)
+
+# from the back
```
Comment (aljoscha) [inline test/sqllogictest/string.slt:979]: Shouldn't this be "from the front"? And later you have "from the back", which should be exchanged.

Comment (antiguru) [reply test/sqllogictest/string.slt:979]: Good catch, my search&replace didn't match it :D

File: test/sqllogictest/string.slt (hunk 1 above)
Comment (antiguru) [reply test/sqllogictest/string.slt:1046]: Yep, same result.

File: doc/user/content/release-notes.md [hunk 3]
```diff
@@ -58,6 +58,8 @@ Wrap your release notes at the 80 character mark.
   `mz_kafka_consumer_partitions` has an additional column `initial_high_offset`
   containing the first reported `hi_offset` from the broker for each partition.
 
+- Add `left` to the [string function](/sql/functions#string-func) suite.
```
Comment (aljoscha) [inline doc/user/content/release-notes.md:61]: Excellent, always be adding release notes. :smiley:

File: src/sql/src/func.rs [hunk 4]
```diff
@@ -1473,6 +1473,9 @@ lazy_static! {
             "jsonb_typeof" => Scalar {
                 params!(Jsonb) => UnaryFunc::JsonbTypeof, 3210;
             },
+            "left" => Scalar {
+                params!(String, Int32) => BinaryFunc::Left, 3060;
```
Comment (aljoscha) [inline src/sql/src/func.rs:1477]: OID confirmed!

**Review by aljoscha — APPROVED** [review]
This looks good to merge! :+1:

**Review by philip-stoev — APPROVED** [review]
Thank you, the tests look comprehensive -- there is nothing for me to add.

**Comment by antiguru** [comment]
Thanks for the reviews!

**Pull request merged**